src/components/vm/vmActions.jsx

```diff
@@ -219,29 +230,93 @@ const VmActions = ({ vm, onAddErrorNotification, isDetailsPage }) => {
                     variant={isDetailsPage ? 'primary' : 'secondary'}
                     isLoading={operationInProgress}
                     isDisabled={operationInProgress}
-                    onClick={() => { setOperationInProgress(true); onShutdown() }} id={`${id}-shutdown-button`}>
+                    id={`${id}-shutdown-button`}
+                    onClick={() => Dialogs.show(
+                        <ConfirmDialog idPrefix={id}
+                            title={fmt_to_fragments(_("Shut down $0"), <b>{vm.name}</b>)}
+                            titleIcon={PowerOffIcon}
+                            vm={vm}
+                            actionsList={[
+                                {
+                                    variant: "primary",
+                                    handler: () => {
+                                        setOperationInProgress(true);
+                                        onShutdown(vm, setOperationInProgress);
+                                    },
+                                    name: _("Shut down"),
+                                    id: "off",
+                                },
+                                {
+                                    variant: "secondary",
+                                    handler: () => {
+                                        setOperationInProgress(true);
+                                        onReboot(vm);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1069-20230621-115207-690ab5dc-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)